cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+					Requests: nil,
+					Claims:   nil,
+				},
+				ResizePolicy:  nil,
+				VolumeMounts:  nil,
+				VolumeDevices: nil,
+				LivenessProbe: nil,
+				ReadinessProbe: &apiv1.Probe{
+					ProbeHandler: apiv1.ProbeHandler{
+						Exec: nil,
+						HTTPGet: &apiv1.HTTPGetAction{
+							Path:        "/ready",
+							Port:        intstr.FromInt(lokiPort),
+							Host:        "",
+							Scheme:      "",
+							HTTPHeaders: nil,
+						},
+						TCPSocket: nil,
+						GRPC:      nil,
+					},
+					InitialDelaySeconds:           lokiProbInitialDelaySeconds,
+					TimeoutSeconds:                lokiProbeTimeoutSeconds,
+					PeriodSeconds:                 lokiProbePeriodSeconds,
+					SuccessThreshold:              0,
+					FailureThreshold:              0,
+					TerminationGracePeriodSeconds: nil,
+				},
+				StartupProbe:             nil,
+				Lifecycle:                nil,
+				TerminationMessagePath:   "",
+				TerminationMessagePolicy: "",
+				ImagePullPolicy:          "",
+				SecurityContext:          nil,
+				Stdin:                    false,
+				StdinOnce:                false,
+				TTY:                      false,
+			},
+		},
+		[]apiv1.Volume{},
+		&apiv1.Affinity{
+			NodeAffinity:    nil,
+			PodAffinity:     nil,
+			PodAntiAffinity: nil,
+		})
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Loki deployment.")
+	}
+	shouldRemoveLokiDeployment := false
+	defer func() {
+		if shouldRemoveLokiDeployment {
+			if err := k8sManager.RemoveDeployment(ctx, graflokiNamespace, lokiDeployment); err != nil {
+				logrus.Warnf("Attempted to remove Loki deployment after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Loki deployment with name: %v", lokiDeployment.Name)
+			}
+		}
+	}()
+	logrus.Infof("Waiting for Loki deployment to come online (can take around 30s)... ")
+	if err := k8sManager.WaitForPodManagedByDeployment(ctx, lokiDeployment, lokiDeploymentMaxRetires, lokiDeploymentRetryInterval); err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred while waiting for pod managed by Loki deployment '%v' to come online.", lokiDeploymentName)
+	}
+
+	lokiService, err := k8sManager.CreateService(ctx,
+		graflokiNamespace,
+		lokiServiceName,
+		map[string]string{}, // empty labels
+		map[string]string{}, // empty annotations
+		lokiLabels,          // match loki deployment pod labels
+		apiv1.ServiceTypeNodePort,
+		[]apiv1.ServicePort{{
+			Name:        "logs-listening",
+			Port:        lokiNodePort,
+			TargetPort:  intstr.FromInt(lokiPort),
+			Protocol:    apiv1.ProtocolTCP,
+			NodePort:    lokiNodePort,
+			AppProtocol: &httpApplicationProtocol,
+		}})
+	shouldRemoveLokiService := false
+	defer func() {
+		if shouldRemoveLokiService {
+			if err := k8sManager.RemoveService(ctx, lokiService); err != nil {
+				logrus.Warnf("Attempted to remove Loki service after an error occurred creating it but an error occurred removing it.")
```

Comment:
Same comment about the language in this (if we got here, we actually _succeeded_ in creating the Loki service)